components/raftstore/src/store/peer.rs

```diff
@@ -1316,8 +1334,6 @@ where
         // could get a newer value, and after that, the leader may read a stale value,
         // which violates linearizability.
         self.get_store().applied_index() >= read_index
-            && !self.is_splitting()
```

Comment:
Why remove this?

Reply:
Because it's useless. `is_splitting` and `is_merging` should only be used on leader for its lease.
Actually, the `last_committed_split_idx` and `last_committed_prepare_merge_idx` are maintained only on leader all the time.